src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.netcoreapp.Unix.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.NetCore.Extensions;
+
+namespace System.IO.Pipes.Tests
+{
+    /// <summary>
+    /// Negative tests for PipeOptions.CurrentUserOnly in Unix.
+    /// </summary>
+    public class NamedPipeTest_CurrentUserOnly_Unix : RemoteExecutorTestBase
+    {
+        [Theory]
+        [OuterLoop("Needs sudo access")]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        [InlineData(PipeOptions.None, PipeOptions.None)]
+        [InlineData(PipeOptions.None, PipeOptions.CurrentUserOnly)]
+        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.None)]
+        [InlineData(PipeOptions.CurrentUserOnly, PipeOptions.CurrentUserOnly)]
+        public async Task Connection_UnderDifferentUsers_BehavesAsExpected(
+            PipeOptions serverPipeOptions, PipeOptions clientPipeOptions)
+        {
+            // Use an absolute path, otherwise, the test can fail if the remote invoker and test runner have
+            // different working and/or temp directories. 
+            string pipeName = "/tmp/" + Path.GetRandomFileName(); 
+            using (var server = new NamedPipeServerStream(
+                pipeName, PipeDirection.InOut, 1, PipeTransmissionMode.Byte, serverPipeOptions | PipeOptions.Asynchronous))
+            {
+                Task serverTask = server.WaitForConnectionAsync(CancellationToken.None);
+
+                using (RemoteInvoke(
+                    new Func<string, string, int>(ConnectClientFromRemoteInvoker),
+                    pipeName,
+                    clientPipeOptions == PipeOptions.CurrentUserOnly ? "true" : "false",
+                    new RemoteInvokeOptions { RunAsSudo = true }))
+                {
+                }
+
+                if (serverPipeOptions == PipeOptions.CurrentUserOnly)
+                    await Assert.ThrowsAsync<UnauthorizedAccessException>(() => serverTask);
```

Comment:
This is testing a user against a sudoer. Is there a test for one regular user vs another regular user? I don't know how sudo actually works, but it's not obvious to me that it's equivalent to testing with a regular user.

If you have sudo powers, you can add a user, and then run as them presumably.

Reply:
There is no such test case, I initially considered going this route but creating the user in macOS seemed a bit convoluted (I may have missed an easy way to do that). Anyway, in this case the super user is equivalent because we use `getpeerid` to validate if the other side of the pipe (the peer) has the same id, so the super user is not special - the production code just need to see a different id values to be tested.